src/gt4py/backend/numpy_backend.py

```diff
@@ -271,23 +270,29 @@ def _visit_branch_stmt(self, stmt: gt_ir.Statement) -> List[str]:
                 if target in self.block_info.symbols
                 else stmt.target.data_type
             )
+            is_possible_else = not isinstance(stmt.target, gt_ir.VarRef) or (
```

Comment:
Can you quickly explain to me how this works? It feels like it's correct but I don't fully understand it :man_facepalming:

Reply:
You call a numpy where. If this is the first time the VarRef is used, it can't be the else-clause of the where since it would be undefined. 
```
b = np.where(a > 1, c, b)
NameError: name 'b' is not defined
```
if b was defined already it need to be the else-part of the if in order to not lose previously computed info. An example for this would be limiters like:
```
snowd = min(snowd, HSMAX)
if snowd > 2.0 * hice:
    snowd = hice + hice
```

Everything that is not a VarRef can by design be on the RHS (like fields) and VarRefs can only if they are in the `var_refs_defined` set)

Reply:
Thanks for the explanation. Everything makes sense now. Let me know if you agree to add a short version of this explanation to the source as a comment before the PR is merged.

Reply:
of course. I can give it a shot